exchanges/kucoin/kucoin.go

```diff
@@ -311,352 +205,213 @@ func (ku *Kucoin) GetKlines(ctx context.Context, pair, period string, start, end
 	if !end.IsZero() {
 		params.Set("endAt", strconv.FormatInt(end.Unix(), 10))
 	}
-	var resp [][7]string
-	err := ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, common.EncodeURLValues(kucoinGetKlines, params), &resp)
+	var resp [][7]types.Number
+	err := ku.SendHTTPRequest(ctx, exchange.RestSpot, klinesEPL, common.EncodeURLValues("/v1/market/candles", params), &resp)
 	if err != nil {
 		return nil, err
 	}
 	klines := make([]Kline, len(resp))
 	for i := range resp {
-		t, err := strconv.ParseInt(resp[i][0], 10, 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].StartTime = time.Unix(t, 0)
-		klines[i].Open, err = strconv.ParseFloat(resp[i][1], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Close, err = strconv.ParseFloat(resp[i][2], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].High, err = strconv.ParseFloat(resp[i][3], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Low, err = strconv.ParseFloat(resp[i][4], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Volume, err = strconv.ParseFloat(resp[i][5], 64)
-		if err != nil {
-			return nil, err
-		}
-		klines[i].Amount, err = strconv.ParseFloat(resp[i][6], 64)
-		if err != nil {
-			return nil, err
-		}
+		klines[i].StartTime = time.Unix(resp[i][0].Int64(), 0)
+		klines[i].Open = resp[i][1].Float64()
+		klines[i].Close = resp[i][2].Float64()
+		klines[i].High = resp[i][3].Float64()
+		klines[i].Low = resp[i][4].Float64()
+		klines[i].Volume = resp[i][5].Float64()
+		klines[i].Amount = resp[i][6].Float64()
 	}
 	return klines, nil
 }
 
-// GetCurrencies gets list of currencies
-func (ku *Kucoin) GetCurrencies(ctx context.Context) ([]Currency, error) {
-	var resp []Currency
-	return resp, ku.SendHTTPRequest(ctx, exchange.RestSpot, defaultSpotEPL, kucoinGetCurrencies, &resp)
+// GetCurrenciesV3 the V3 of retrieving list of currencies
+func (ku *Kucoin) GetCurrenciesV3(ctx context.Context) ([]CurrencyDetail, error) {
```

Comment:
`CurrencyDetail` is lacking many of the response details in: https://www.kucoin.com/docs/rest/spot-trading/market-data/get-currency-detail